app/models/host_status/execution_status.rb

```diff
@@ -37,6 +41,47 @@ def to_label(options = {})
     end
   end
 
+  class ExecutionTaskStatusMapper
+    attr_accessor :task
+
+    def self.sql_conditions_for(status)
+      status = STATUS_NAMES.key(status) if status.is_a?(String)
+
+      case status
+        when OK
+          [ "state = 'stopped' AND (result = 'success' OR result = 'warning')" ]
```

Comment:
result `warning` probably deserves separate state: It's a bit strange for a job with some failed hosts to be marked `success`, possible variants `warning` or `partial sucess` or something else?